guides/source/action_controller_overview.md

```diff
@@ -117,109 +124,146 @@ NOTE: The `params` hash is not a plain old Ruby Hash; instead, it is an [`Action
+When this form is submitted, the value of `params[:user]` will be `{ "name" => "Acme", "phone" => "12345", "address" => { "postcode" => "12345", "city" => "Carrot City" } }`. Note the nested hash in `params[:user][:address]`.
 
 The `params` object acts like a Hash, but lets you use symbols and strings interchangeably as keys.
 
 ### JSON Parameters
 
-If your application exposes an API, you are likely to be accepting parameters in JSON format. If the "Content-Type" header of your request is set to "application/json", Rails will automatically load your parameters into the `params` hash, which you can access as you would normally.
+If your application exposes an API, you will likely accept parameters in JSON format. If the "Content-Type" header of your request is set to "application/json", Rails will automatically load your parameters into the `params` hash, which you can access as you would normally.
 
 So for example, if you are sending this JSON content:
 
 ```json
-{ "company": { "name": "acme", "address": "123 Carrot Street" } }
+{ "user": { "name": "acme", "address": "123 Carrot Street" } }
 ```
 
-Your controller will receive `params[:company]` as `{ "name" => "acme", "address" => "123 Carrot Street" }`.
+Your controller will receive `params[:user]` as `{ "name" => "acme", "address" => "123 Carrot Street" }`.
 
-Also, if you've turned on `config.wrap_parameters` in your initializer or called [`wrap_parameters`][] in your controller, you can safely omit the root element in the JSON parameter. In this case, the parameters will be cloned and wrapped with a key chosen based on your controller's name. So the above JSON request can be written as:
+#### Configuring Wrap Parameters
+
+You can use [Wrap Parameters][], which automatically add the controller name to JSON parameters. For example, you can send the below JSON without a root `:user` key prefix:
 
 ```json
 { "name": "acme", "address": "123 Carrot Street" }
 ```
 
-And, assuming that you're sending the data to `CompaniesController`, it would then be wrapped within the `:company` key like this:
+Assuming that you're sending the above data to the `UsersController`, the JSON will be wrapped within the `:user` key like this:
 
 ```ruby
-{ name: "acme", address: "123 Carrot Street", company: { name: "acme", address: "123 Carrot Street" } }
+{ name: "acme", address: "123 Carrot Street", user: { name: "acme", address: "123 Carrot Street" } }
 ```
 
-You can customize the name of the key or specific parameters you want to wrap by consulting the [API documentation](https://api.rubyonrails.org/classes/ActionController/ParamsWrapper.html)
+NOTE: Wrap Parameters clone the parameters and add the controller name as a key. The "naked" parameters still exist in the `params` hash.
```

Comment:
Just a nit here:

I think this could maybe be a tad more clear. Something to the effect of:

> NOTE: Wrap Parameters _adds_ a clone of the parameters within the controller name key. As a result, both the original version of the parameters and the "wrapped" version of the parameters will exist in the `params` hash.